feature: Create Azure Terraform modules. Closes: cksource/cs#8996 (#8)
cksource/ckeditor-cs-on-premises-infrastructure: azure/aks/example-usage/README.md

> # Example usage of Azure Terraform modules

Reviewer: `diagram.jpg` it's a little bit messy IMO. I mean, when you use a diagram after a first look you should know what is happening. On this diagram, you have to analyze the diagram.

Author: Will try to do sth about it

> This catalog contains example usage of Terraform modules prepared for
> provisioning CKEditor Collaboration Server.

Reviewer 2: ```suggestion
This catalog contains examples of how the Terraform modules can be used for provisioning Collaboration Server On-Premises.
```

> **Note**: Before starting you have to login into the Azure CLI tool. You can

Reviewer 2: ```suggestion
**Note**: Before starting, you have to log into the Azure CLI tool. You can
```

download it here: https://docs.microsoft.com/en-us/cli/azure/

> ## Quickstart
> The project requires resource group `ckeditor-cs` and storage account in that
> resource group named the same - `ckeditor-cs`. You can change used resource group
> and storage group inside [infrastructure/main.tf](./infrastructure/main.tf) file
> and [service/main.tf](./service/main.tf)
> 
> The first step is to provide the infrastructure needed to run the service. The
> the process can take a long duration of time, usually around 30 minutes.
> 
> ```bash
> cd infrastructure
> 
> # Initialize the terraform project
> terraform init
> 
> # Verify resources planned to create and apply their creation
> terraform apply
> ```
> 
> After the creation of the infrastructure, it is time to apply the service
> module.

Reviewer 2: ```suggestion
## Quickstart
The project requires a resource group called `ckeditor-cs` and a storage account in that
resource group named the same - `ckeditor-cs`. You can change the used resource group
and the storage group in [infrastructure/main.tf](./infrastructure/main.tf) and [service/main.tf](./service/main.tf)

The first step is to provide the infrastructure needed to run the service. The
the process can take a long time, usually around 30 minutes.

```bash
cd infrastructure

# Initialize the terraform project
terraform init

# Verify the resources planned to be created and apply their creation
terraform apply
```

Once the infrastructure has been created, it is the time to apply the service
module.
```

Author: Applied it manually, because of the problem of escaping ` ``` `

> **Note**: While running the `apply` command you will be required to pass your
> `docker_token`, `license_key` and `environments_management_secret_key` Where to
> find your credentials:
> - The `license_key` can be found in [CKEditor Ecosystem
>   Dashboard](https://dashboard.ckeditor.com/) in your CKEditor Collaboration
>   Server On-Premises subscription page.
> - The `docker_token` can be found in [CKEditor Ecosystem
>   Dashboard](https://dashboard.ckeditor.com/) in your
>   CKEditor Collaboration Server On-Premises subscription page in the *Download
>   token* section. If you do not see any tokens, you can create them with the
>   *Create new token* button.
> - The `environments_management_secret_key` is your password, that will be used
>   to access the Collaboration Server On-Premises [Management
>   Panel](https://ckeditor.com/docs/cs/latest/onpremises/cs-onpremises/management.html)
> 
> 
> **Note 2**: You can omit using terraform module to provision the CKEditor
> Collaboration Server by using Helm directly. Check our Helm charts directory for
> more information: [Helm](/kubernetes/helm/)

Reviewer 2: ```suggestion
**Note**: While running the `apply` command, you will be required to pass your
`docker_token`, `license_key` and `environments_management_secret_key`. Here is where to
find your credentials:
- `license_key` can be found in the [CKEditor Ecosystem
  Dashboard](https://dashboard.ckeditor.com/) on your CKEditor Collaboration
  Server On-Premises subscription page.
- `docker_token` can be found in the [CKEditor Ecosystem
  Dashboard](https://dashboard.ckeditor.com/) on your
  CKEditor Collaboration Server On-Premises subscription page in the *Download
  token* section. If you do not see any tokens, you can create them with the
  *Create new token* button.
- `environments_management_secret_key` is your password used
  to access the Collaboration Server On-Premises [Management
  Panel](https://ckeditor.com/docs/cs/latest/onpremises/cs-onpremises/management.html)

**Note 2**: You can omit using Terraform modules to provision Collaboration Server On-Premises by using Helm directly. Check our Helm charts directory for more information: [Helm](/kubernetes/helm/)
```

```bash
cd service

# Initialize the terraform project
terraform init

# Verify resources planned to create and apply their creation
terraform apply
```

> The CKEditor Collaboration Server should be accessible under the IP address
> assigned to created Application Gateway. You can find the IP of the gateway in
> the Azure panel or by Azure CLI.
> 
> You can validate the deployment of the CKEditor Collaboration Server by
> accessing the management panel under the `/panel` path. So, it will be
> `http://${APPLICATION_GATEWAY_FRONTEND_IP}/panel`.

Reviewer 2: ```suggestion
Collaboration Server On-Premises should be accessible under the IP address
assigned to the created Application Gateway. You can find the IP of the gateway in
the Azure panel or via the Azure CLI.

You can validate the deployment of the Collaboration Server On-Premises by accessing the management panel under the `/panel` path, i.e. `http://${APPLICATION_GATEWAY_FRONTEND_IP}/panel`.
```

# Example usage of Azure Terraform modules

This catalog contains examples of how the Terraform modules can be used for
provisioning Collaboration Server On-Premises.

**Note**: Before starting, you have to log into the Azure CLI tool. You can
download it here: https://docs.microsoft.com/en-us/cli/azure/

## Quickstart
The project requires a resource group called `ckeditor-cs` and a storage account
in that resource group named the same - `ckeditor-cs`. You can change the used
resource group and the storage group in
[infrastructure/main.tf](./infrastructure/main.tf) and
[service/main.tf](./service/main.tf)

The first step is to provide the infrastructure needed to run the service. The
the process can take a long time, usually around 30 minutes.

> **Note**: While running the `apply` command, you will be required to pass your
> `ssh_public_key_path`. Content of the file under the path will be used as a
> authorized ssh key to connect to nodes in Kubernetes cluster.

Reviewer 2: ```suggestion
**Note**: While running the `apply` command, you will be required to pass your
`ssh_public_key_path`. The content of the file under the path will be used as
an authorized SSH key to connect to the nodes in the Kubernetes cluster.
```
